requirements/base.txt

```diff
@@ -147,6 +147,37 @@ numpy==1.26.4
     #   torchvision
     #   transformers
     #   unstructured
+nvidia-cublas-cu12==12.1.3.1
+    # via
+    #   nvidia-cudnn-cu12
+    #   nvidia-cusolver-cu12
+    #   torch
+nvidia-cuda-cupti-cu12==12.1.105
+    # via torch
+nvidia-cuda-nvrtc-cu12==12.1.105
+    # via torch
+nvidia-cuda-runtime-cu12==12.1.105
+    # via torch
+nvidia-cudnn-cu12==8.9.2.26
+    # via torch
+nvidia-cufft-cu12==11.0.2.54
+    # via torch
+nvidia-curand-cu12==10.3.2.106
+    # via torch
+nvidia-cusolver-cu12==11.4.5.107
+    # via torch
+nvidia-cusparse-cu12==12.1.0.106
+    # via
+    #   nvidia-cusolver-cu12
+    #   torch
+nvidia-nccl-cu12==2.19.3
+    # via torch
+nvidia-nvjitlink-cu12==12.3.101
+    # via
+    #   nvidia-cusolver-cu12
+    #   nvidia-cusparse-cu12
+nvidia-nvtx-cu12==12.1.105
+    # via torch
```

Comment:
Hmm not sure why did it appear. It says "via torch", but torch was already there. Looking into torch 2.2.1 changelog https://github.com/pytorch/pytorch/releases I don't see anything about those libs.

Reply:
I believe we've seen this happen when your pyenv is the wrong version. Does it go away when you explicitly use 3.10? Might even want to try 3.9.

Reply:
Hmm but I'm using 3.10:

> $ which pip-compile
> /home/hubert/.pyenv/versions/3.10.13/envs/unstructured-api/bin/pip-compile

When it was global pip-compile before, it was even generating it for python 3.8 and the changes were worse, ie. some libs were downgraded.

Reply:
Ok so I spent a lot of time debugging this and can't get rid of the additional nvidia libs. From what I've heard from guys from enterprise team, they also have similar problems on Linux. The solution they use is to have the libs be pip-compiled by someone owning a Mac. :apple: wins this round. @awalker4 could you help with this? I have pushed to branch a merge from main and reset the dependencies to main, which means in this state tests will fail as `deepdiff` is not installed.

Reply:
Yep, let me push a commit soon